automod/triggers.go

```diff
@@ -1087,7 +1133,7 @@ func (r *NicknameRegexTrigger) CheckNickname(t *TriggerContext) (bool, error) {
 	}
 
 	re := item.Value().(*regexp.Regexp)
-	if re.MatchString(t.MS.Member.Nick) {
+	if re.MatchString(t.MS.Member.Nick) || dataCast.SanitizeText && re.MatchString(common.SanitizeText(t.MS.Member.Nick)) {
```

Comment:
don't make conditions unnecessarily complex, store computed values in a variable and to make it readable, I don't want to guess what you're trying to do.